### ISSUE TITLE
lightning-sys should be able to use already-installed lightning

### ISSUE THREAD

**Issue by kulp** [Petelliott/lightning-sys#21]
Using a "batteries-included" approach, `lightning-sys` downloads and installs a particular version of GNU Lightning. It seems likely that some users may wish to link against an already-installed version of GNU Lightning.

I suggest that we add a [feature](https://doc.rust-lang.org/cargo/reference/features.html) to the `Cargo.toml` manifest, perhaps called `build-own-lightning` or some such, to permit a user or crate to avoid downloading, compiling, and linking against the Lightning version we ourselves choose. We would also be able to make some dependencies like `reqwest` dependent on the enabling of that feature.

For backward compatibility and user-friendliness, it may be best to enable the new feature by default.

**Comment by Petelliott** [comment]
I would like this very much.
I think in the spirit of Cargo's npm style package management, each version should always be built against the same version of lightning, so the default should be download and build.

**Comment by kulp** [comment]
Keeping the current behavior as default sounds fine to me.

I intend to get to this eventually, but it will probably not be what I do first. I am working on setting up coverage testing with `tarpaulin` at present.

**Comment by kulp** [comment]
I just realized that since #10 switched to `dylib` linking for `liblightning`, we have a big problem -- an external user of `lightning-sys` will not know where to find the dynamic `liblightning` (such as `liblightning.so.1` on Linux, or `liblightning.dylib` on macOS), because it exists only in the crate build directory. This can manifest like this:
```
<name-of-binary>: error while loading shared libraries: liblightning.so.1: cannot open shared object file: No such file or directory
```

I noticed this because even `cargo tarpaulin` does not find `liblightning.so.1` unless it happens to be installed on the host computer already.

This was an unintended consequence of #10 and so I am sorry for that. I think static linking would be best, for multiple reasons, but I do not yet have an idea how to make it work given the issues raised in e9301f4b2a4f544ecb8faaf76bd9be1c8a320d0d.

We could reinstate the workaround from c465a462 (removed in 522ee665), but that is a bit of a hack.

Unless I hear differently, I will soon open a new issue to track this problem.

**Comment by Petelliott** [comment]
> We could reinstate the workaround from [c465a46](https://github.com/Petelliott/lightning-sys/commit/c465a4628bafaa79db31de3683564c142f05ed12) (removed in [522ee66](https://github.com/Petelliott/lightning-sys/commit/522ee665afd9659cd0c26b0f7847c76f91de2b85)), but that is a bit of a hack.

I don't think that ever actually worked. (execpt maybe specifically on rust 1.36.0). it was removed because the `recompile -fPIC` warning still occured

**Comment by kulp** [comment]
[A helpful web page I found](https://kornel.ski/rust-sys-crate) suggests that the [`pkg-config` crate](https://crates.io/crates/pkg-config) with static linking could cover most major operating systems except Windows. I will try to follow that site's recommendations when I resolve the current issue.
